src/Core/maps/src/Platform/iOS/MauiMKMapView.cs

```diff
@@ -0,0 +1,305 @@
+		{
+			MKAnnotationView? mapPin;
+
+			// https://bugzilla.xamarin.com/show_bug.cgi?id=26416
+			var userLocationAnnotation = Runtime.GetNSObject(annotation.Handle) as MKUserLocation;
+			if (userLocationAnnotation != null)
+				return null!;
+
+			const string defaultPinId = "defaultPin";
+			mapPin = mapView.DequeueReusableAnnotation(defaultPinId);
+			if (mapPin == null)
+			{
+				if (OperatingSystem.IsIOSVersionAtLeast(11))
+				{
+					mapPin = new MKMarkerAnnotationView(annotation, defaultPinId);
+				}
+				else
+				{
+					mapPin = new MKPinAnnotationView(annotation, defaultPinId);
+
+				}
+
+				mapPin.CanShowCallout = true;
+
+				if (OperatingSystem.IsIOSVersionAtLeast(11))
+				{
+					// Need to set this to get the callout bubble to show up
+					// Without this no callout is shown, it's displayed differently
+					mapPin.RightCalloutAccessoryView = new UIView();
+				}
+			}
+
+			mapPin.Annotation = annotation;
+			AttachGestureToPin(mapPin, annotation);
+
+			return mapPin;
+		}
+
+		void AttachGestureToPin(MKAnnotationView mapPin, IMKAnnotation annotation)
+		{
+			var recognizers = mapPin.GestureRecognizers;
+
+			if (recognizers != null)
+			{
+				foreach (var r in recognizers)
+				{
+					mapPin.RemoveGestureRecognizer(r);
+				}
+			}
+
+			var recognizer = new UITapGestureRecognizer(g => OnCalloutClicked(annotation))
+			{
+				ShouldReceiveTouch = (gestureRecognizer, touch) =>
+				{
+					_lastTouchedView = touch.View;
+					return true;
+				}
+			};
+
+			mapPin.AddGestureRecognizer(recognizer);
+		}
+
+		void OnCalloutClicked(IMKAnnotation annotation)
+		{
+			// lookup pin
+			var targetPin = GetPinForAnnotation(annotation);
+
+			// pin not found. Must have been activated outside of forms
+			if (targetPin == null)
+				return;
+
+			// if the tap happened on the annotation view itself, skip because this is what happens when the callout is showing
+			// when the callout is already visible the tap comes in on a different view
+			if (_lastTouchedView is MKAnnotationView)
+				return;
+
+			targetPin.SendMarkerClick();
+
+			// SendInfoWindowClick() returns the value of PinClickedEventArgs.HideInfoWindow
+			// Hide the info window by deselecting the annotation
+			bool deselect = targetPin.SendInfoWindowClick();
+			if (deselect)
+			{
+				DeselectAnnotation(annotation, true);
+			}
+		}
+
+		internal void AddPins(IList pins)
+		{
+			if (_handler?.MauiContext == null)
+				return;
+
+			if (Annotations?.Length > 0)
+				RemoveAnnotations(Annotations);
+
+			foreach (IMapPin pin in pins)
+			{
+				if (pin.ToHandler(_handler.MauiContext).PlatformView is IMKAnnotation annotation)
+				{
+					pin.MarkerId = annotation;
+					AddAnnotation(annotation);
+				}
+			}
+		}
+
+		internal void AddElements(IList elements)
+		{
+			foreach (IMapElement element in elements)
+			{
+				//	element.PropertyChanged += MapElementPropertyChanged;
```

Comment:
So in iOS i didn't implemented when the MapElement changes properties, since these are different objects from what the MapElement handler creates

Reply:
Before what the old renderer did was add remove the element when a property changed.. @PureWeen  any idea how i could plug that here ?

Reply:
Isn't this now handled by the handlers since each `MapElement` has its own handler so the handler will react to property changes and then propagate those out/up as necessary?

Reply:
There are 2 types of objects that goes to a MapElement, Theres a `IMKOverlay`  that is added here on the `MauiMKMApView`, and then theres a `MKOverlayRenderer`, and that's what the handler returns. 

When a property changes like the Locations on `MKPolygon`, i need to update the `IMKOverlay` not the `MKOverlayRenderer`.

Reply:
I think I'm following here

So, I'm wondering if this would work via the CommandMapper

![image](https://user-images.githubusercontent.com/5375137/184926092-0c4395ba-e598-42ee-b8c7-ac8b30273c33.png)

or call it `MapElementChanged` or similar.

Then the args on that would pass in the element that's changed?